front-vite-react/front-vite-react/src/Pages/Calendar.jsx

```diff
@@ -1,115 +1,78 @@
-import React, { useState, useEffect } from 'react';
-import { Link } from 'react-router-dom';
+import React, { useEffect } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { motion } from 'framer-motion'; // Importando motion do framer-motion
-import './Css/Calendar.css';
-
-const CalendarHeader = ({ currentDate, prevMonth, nextMonth }) => {
-    return (
-        <motion.div
-            className="header"
-            initial={{ opacity: 0, y: -20 }}
-            animate={{ opacity: 1, y: 0 }}
-            transition={{ duration: 0.5 }}
-        >
-            <button className="nav-btn" onClick={prevMonth}>&lt;</button>
-            <h2 className="month">{currentDate.toLocaleString('default', { month: 'long', year: 'numeric' })}</h2>
-            <button className="nav-btn" onClick={nextMonth}>&gt;</button>
-        </motion.div>
-    );
-};
-
-const DayNames = () => {
-    const dayNames = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
-    return (
-        <motion.div
-            className="day-names"
-            initial={{ opacity: 0, y: -20 }}
-            animate={{ opacity: 1, y: 0 }}
-            transition={{ duration: 0.5, delay: 0.2 }}
-        >
-            {dayNames.map((day, index) => (
-                <div key={index} className="day-name">{day}</div>
-            ))}
-        </motion.div>
-    );
-};
+import FullCalendar from '@fullcalendar/react';
+import dayGridPlugin from '@fullcalendar/daygrid';
+import { motion } from 'framer-motion';
 
-const Day = ({ day }) => {
-    return (
-        <Link to={`/day/${day}`} className="day-link">
-            <motion.div
-                className="day"
-                initial={{ opacity: 0, scale: 0.5 }}
-                animate={{ opacity: 1, scale: 1 }}
-                transition={{ duration: 0.5, delay: 0.4 }}
-            >
-                {day}
-            </motion.div>
-        </Link>
-    );
-};
+import './Css/Calendar.css';
 
 const Calendar = () => {
-    const [currentDate, setCurrentDate] = useState(new Date());
     const navigate = useNavigate();
 
     useEffect(() => {
         const token = localStorage.getItem('token');
```

Comment:
Ao invés de verificar se tem um token ou não salvo no `localStorage` através de um `useEffect`, use um context global/middleware ou simplesmente uma store. Listo aqui algumas alternativas:

Context do React: https://react.dev/reference/react/useContext
Jotai: https://jotai.org/
Zustand: https://docs.pmnd.rs/zustand/getting-started/introduction

Embora seja comum também se achar exemplos por ai onde se fazem algum `fetch` ou tratamento de efeitos colaterais usando o `useEffect`, é considerado uma má prática utilizar ele e devemo evitar ao máximo. Deixo aqui alguns materiais de referência sobre esse tema:

Postagem de Dan Abramov (um dos criadores do React) falando tudo sobre o useEffect: https://overreacted.io/a-complete-guide-to-useeffect/

Esse vídeo mostra caminhos alternativos ao useEffect: https://www.youtube.com/watch?v=Rub-XMihzis

A `fetch` que tem nesse componente aparenta estar fazendo nada... De qualquer forma, vi que nas dependências do projeto o [react-query](https://www.npmjs.com/package/react-query) está listado. Como consta na própria documentação do pacote no NPM, ele foi descontinuado e agora se encontra como [@tanstack/react-query](https://www.npmjs.com/package/@tanstack/react-query). Aconselho fortemente que troquem para a versão mais nova (no geral não muda muitas API's)

![image](https://github.com/ArielAioES/ProjetoAgenda/assets/117988687/2ea4300d-5b3d-4952-9bcc-f804ba16e262)